packages/wallet-sdk/src/core/provider/interface.ts

```diff
@@ -42,10 +42,47 @@ export interface AppMetadata {
   appChainIds: number[];
 }
 
-export interface Preference {
+type PostOnboardingAction = 'none' | 'onramp' | 'magicspend';
+
+type OnrampPrefillOptions = {
+  contractAddress?: string;
+  amount: string;
+  chainId: number;
+};
+
+export type Preference = {
   options: 'all' | 'smartWalletOnly' | 'eoaOnly';
+  /**
+   * @deprecated this is only for internal use
+   */
   keysUrl?: string;
-}
+  /**
+   * @param postOnboardingAction
+   * @type {PostOnboardingAction}
+   * @description Recommends the action to be taken after the onboarding process to engage users with various flows. Possible values:
```

Comment:
> Recommends the action to be taken after the onboarding process to engage users with various flows. Possible values:

I think we could add some clarity here
"Displays CTAs to the user based on the preference of the dapp. These CTAs are part of prebuilt UI components that are available to the CB Smart Wallet"

Reply:
Thanks good callout